### ISSUE TITLE
Add identity-based DISTINCT actor

### ISSUE THREAD

**Issue by rubensworks** [comunica/comunica#52]
The current DISTINCT implementation works based on hashes. ~~While the chance of clashes is quite small~~, we should add an identity-based implementation as well.
**EDIT:** The chance of clashes became higher (#1510), and this is causing issues now.

This could work by for example just stringifying each object (e.g. using rdf-string), instead of hashing it.

We should add this in a new `actor-query-operation-distinct-identity` actor, as alternative to `actor-query-operation-distinct-hash`.

-----

## Bounty

A [bounty](https://comunica.dev/association/bounties/) has been placed on this issue by:

| |
|-|
| [<img alt="Comunica Association" src="https://comunica.dev/img/comunica_red.svg" height="50">](https://comunica.dev/association/) |
| €544 |

[Click here to learn more](https://comunica.dev/association/bounty_process/) if you're interested in claiming this bounty by resolving this issue.

**Comment by rubensworks** [comment]
A bounty has been placed on this issue via the Comunica Association (see original post).

**Comment by milafrerichs** [comment]
> As discussed with @rubensworks, I will work on this issue.

**Comment by rubensworks** [comment]
^ FYI @karelklima.

**Comment by ddeboer** [comment]
With this change, I’m getting duplicate quads again. Just so I understand correctly, does this require using a `DISTINCT` clause in my query?

**Comment by ddeboer** [comment]
You’re right, @karelklima, adding `distinctConstruct` (`--context '{"distinctConstruct": true}'` in the CLI) solves it:

```
comunica-sparql-file --context '{"distinctConstruct": true}' registry.ttl \
                                                                                                        'PREFIX dcat: <http://www.w3.org/ns/dcat#> construct  where {?s a dcat:Dataset ; dcat:distribution ?d . ?d dcat:accessURL ?a }'
```

**Comment by rubensworks** [comment]
This change had nothing to do with `distinctConstruct` AFAIK. So distinct being applied to construct queries without `distinctConstruct` being set to true in the past was definitely unexpected behaviour. The current behaviour of requiring `distinctConstruct` is correct.
This means #1590 can be closed (I hadn't read the issue that well yet).
